qiskit_nature/second_q/operators/electronic_integrals.py

```diff
@@ -0,0 +1,482 @@
+        """Returns only the two-body integrals."""
+        alpha: PolynomialTensor = None
+        if not self.alpha.is_empty() and "++--" in self.alpha:
+            alpha = PolynomialTensor(
+                {"++--": self.alpha["++--"]},
+                validate=False,
+            )
+        beta: PolynomialTensor = None
+        if not self.beta.is_empty() and "++--" in self.beta:
+            beta = PolynomialTensor(
+                {"++--": self.beta["++--"]},
+                validate=False,
+            )
+        mixed: PolynomialTensor = None
+        if not self.mixed.is_empty() and "++--" in self.mixed:
+            mixed = PolynomialTensor(
+                {"++--": self.mixed["++--"]},
+                validate=False,
+            )
+        return ElectronicIntegrals(alpha, beta, mixed)
+
+    @property
+    def register_length(self) -> int | None:
+        """The register length of the internal
+        :class:`qiskit_nature.second_q.operators.PolynomialTensor` instances."""
+        return self.alpha.register_length
+
+    @register_length.setter
+    def register_length(self, reg_length: int | None) -> None:
+        self.alpha.register_length = reg_length
+
+    def __getitem__(self, __k: str) -> PolynomialTensor:
+        try:
+            return self.__getattribute__(__k)
+        except AttributeError as exc:
+            raise KeyError from exc
+
+    def __iter__(self) -> Iterator[str]:
+        for key in "alpha", "beta", "mixed":
+            yield key
+
+    def __eq__(self, other: object) -> bool:
+        """Check equality of first ElectronicIntegrals with other
+
+        Args:
+            other: second ``ElectronicIntegrals`` object to be compared with the first.
+        Returns:
+            True when ``ElectronicIntegrals`` objects are equal, False when unequal.
+        """
+        if not isinstance(other, PolynomialTensor):
+            return False
+
+        if self.alpha != other.alpha:
+            return False
+
+        if self.beta != other.beta:
+            return False
+
+        if self.mixed != other.mixed:
+            return False
+
+        return True
+
+    def equiv(self, other: object) -> bool:
+        """Check equivalence of first ElectronicIntegrals with other
+
+        Args:
+            other: second ``ElectronicIntegrals`` object to be compared with the first.
+        Returns:
+            True when ``ElectronicIntegrals`` objects are equivalent, False when not.
+        """
+        if not isinstance(other, ElectronicIntegrals):
+            return False
+
+        if not self.alpha.equiv(other.alpha):
+            return False
+
+        if not self.beta.equiv(other.beta):
+            return False
+
+        if not self.mixed.equiv(other.mixed):
+            return False
+
+        return True
+
+    def _multiply(self, other: complex) -> ElectronicIntegrals:
+        return ElectronicIntegrals(
+            cast(PolynomialTensor, other * self.alpha),
```

Comment:
I added the type check :+1: 

As per the `cast`: mypy is not able to detect that `PolynomialTensor` has a `__mul__`/`__rmul__` implementation.. I am not sure if that is because of how the Mixins work or whether mypy just does not bother checking for such a case and defaults to assuming that `number * something` must return a `number`..

Reply:
`mypy` is also not clever enough to infer that `self.alpha * complex` is of type `PolynomialTensor`. It still thinks it must be complex...